src/hocon_parser.yrl

```diff
@@ -17,23 +17,23 @@ Terminals
   include key required.
 
 Rootsymbol hocon.
-hocon -> '{' fields endobj : make_object('$2').
-hocon -> fields : make_object('$1').
+hocon -> '{' fields endobj : make_object(0, '$2').
+hocon -> fields : make_object(0, '$1').
```

Comment:
Now root object has a line number 0 and `make_object` becomes more simple.

However, I am not sure if it makes sense to `try to take the first field’s line number for the object’s line number if 0 is passed in` as you've said. Because I do not imagine any usecases that would need a line number for root object. Obviously a root object does not cause `resolve_error`, and when it comes to `concat_error`, sometimes a root object can be concatenated like this:
```
a = {include "foo.conf}
b = "xyz"${a}
```
but in this case, we want to print the location for `b`, which is `line 2`. 

This is why I propose we leave it to `0`.